src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -237,6 +237,24 @@ static int mainboard_smbios_type16(DMI_INFO *agesa_dmi, int *handle,
 	return len;
 }
 
+static int ddr_speed_from_bus_speed(int bus)
+{
+	switch(bus)
+	{
+		case 166:  return 333;
+		case 216:  return 433;
+		case 266:  return 533;
+		case 333:  return 667;
+		case 556:  return 1111;
+		case 667:  return 1333;
+		case 688:  return 1375;
+		case 813:  return 1625;
+		case 1066: return 2133;
+		case 1666: return 3333;
```

Comment:
@Sakoram don't you see any pattern? It is either `bus*2` or `bus*2 + 1` so you may use following notation:
```
                case 166: 
		case 216: 
		case 266:
		case 333:
		case 556:
		case 667:
		case 688:
		case 813:
		case 1066:
		case 1666:return bus * 2 + 1;
```

Reply:
Actually, some of them were bus*2-1. I've corrected it.